src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -200,11 +146,39 @@ private static HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary
             return response;
         }
 
-        protected async Task<HttpResponse> ExecuteAsync(
+        private HttpClient GetHttpClient(X509Certificate2 x509Certificate2)
+        {
+            if (_httpClientFactory is IMsalMtlsHttpClientFactory msalMtlsHttpClientFactory)
```

Comment:
Since the default client is now an implementation of `IMsalMtlsHttpClientFactory`, this is always true but the cert is not always there. You are adding that check later in the flow. Can this be updated to check if the cert is here and then make a call to either? Also adding a check here will make it more obvious which client is used in which case.

Reason: for the service fabric cert validation, I am trying to create a client with handler. As per the discussion we had, my understanding is if the binding cert is there then we use the MtlsClient and not a default handler.

Reply:
Discussed this with Bogdan and if the service fabric is used then server cert validation will be done and exception will be thrown in case an MTLS cert is used.